compact.go

```diff
@@ -87,10 +95,13 @@ func newCompactorMetrics(r prometheus.Registerer) *compactorMetrics {
 }
 
 type compactorOptions struct {
-	maxBlockRange uint64
+	blockRanges []int64
 }
 
 func newCompactor(dir string, r prometheus.Registerer, l log.Logger, opts *compactorOptions) *compactor {
+	if opts.blockRanges == nil {
+		opts.blockRanges = defaultBlockRanges
+	}
```

Comment:
> What happens if the chosen ranges do not work with already existing blocks?

Some blocks which cross the boundaries of the max-block-range would never be compacted. Which is okay, I guess.